requests-for-comments/2022-06-01-session-analysis.md

```diff
@@ -0,0 +1,118 @@
+
+![image](https://user-images.githubusercontent.com/4813045/171521247-45dfc5d4-d2c5-4ca4-9fec-f81cc0ace5e8.png)
+
+
+### 2. Average session length over time
+
+Proposal: We allow users to do property aggregations based on “session length”
+
+![image](https://user-images.githubusercontent.com/4813045/171521195-121300f6-48a9-488c-956c-f176243865a7.png)
+
+Details:
+- In addition to showing event properties in the taxonomic filter when a property aggregation is selected, we also show “session properties”. Within session properties, we would show session length.
+- If a session property is the target of the property aggregation, then the query would be aggregated by sessions.
+
+### 3. Average number of sessions per user over time
+
+Proposal: Users accomplish this using formulas. They would make a series that is (a) Session count and (b) user count, and then divide (a) by (b).
+
+![image](https://user-images.githubusercontent.com/4813045/171521323-74e94328-1b21-45c8-925f-d9cc3aab6725.png)
+
+### 4. Session length distributions
+
+Proposal: We add session length as an option in the breakdown properties. Additionally, we allow users to “auto-bucket” breakdowns that are on numeric properties. 
+
+![image](https://user-images.githubusercontent.com/4813045/171521347-8f3fd16c-28ab-49ac-9baa-c02097d4c42c.png)
+
+Details: The “auto-bucket” functionality would require more work to figure out exactly how this should work + what the UX should be.
+
+### 5. Count of a specific event per session over time
+
+Proposal: Users accomplish this using formulas. They would make a series that is (a) Session count and (b) total count, and then divide (b) by (a).
+
+![image](https://user-images.githubusercontent.com/4813045/171521365-2b353bb0-7634-45fa-84e3-1463eb351fc0.png)
+
+### 6. Limit a funnel conversion to a single session
+
+Proposal: In funnels, we add “Unique sessions” as an aggregation option. Setting this value would enforce that the sequence of events must occur within a specific session.
+
+![image](https://user-images.githubusercontent.com/4813045/171521401-3b7a234c-24b4-45cd-a3d0-15ab4243efb0.png)
+
+
+### 7. Limit a path conversion to a single session
+
+Proposal: I’d consider this to be a lower priority, but we should update the logic in paths to respect the concept of a session. Today, it uses a 30 min gap between events to split paths - which is essentially the same as sessions, but subtly different.
+
+One issue that we will have to figure out is being backward compatible to data that was generated prior to session support.
+
+## Work breakout:
+
+To accomplish the above, we would need to do the following work:
+
+- Trends:
+    - Add “unique sessions” as an event aggregation option in trends
+    - Add the concept of “session properties”, and specifically the “session length” property to the taxonomic filter.
+    - Allow for session properties to be used as a property aggregation in trends.
```

Comment:
Right now, we only support event properties for property aggregations, there aren't group properties there. For this feature specifically, we would need to extend this to include session length. I think the decision on whether this means we're adding all group properties here or not depends on how we're implementing session length (specifically, is session length a "group property" or something we're calculating on the fly)

<img width="665" alt="image" src="https://user-images.githubusercontent.com/4813045/171684217-841ffd2a-0378-4d12-ade1-4b3b99fe5f4d.png">

Reply:
Instead of special-casing this, let's build out general groups support for both. By "internally" treating sessions as groups we're able to enhance both use-cases.